src/client/microsoftAuth.js

```diff
@@ -11,113 +10,61 @@ const getFetchOptions = {
   }
 }
 
-/**
- * Obtains Minecaft profile data using a Minecraft access token and starts the join sequence
- * @param {object} client - The client passed to protocol
- * @param {object} options - Client Options
- * @param {string} mcAccessToken - Minecraft access token for session server
- * @param {object?} msa - Cached Microsoft account data for more descriptive errors
- */
-async function postAuthenticate (client, options, mcAccessToken, msa) {
-  options.haveCredentials = mcAccessToken != null
-
-  let minecraftProfile
-  const res = await fetch(authConstants.MinecraftServicesProfile, getFetchOptions)
-  if (res.ok) { // res.status >= 200 && res.status < 300
-    minecraftProfile = await res.json()
-  } else {
-    const user = msa ? msa.getUsers()[0] : options.username
-    throw Error(`Failed to obtain Minecraft profile data for '${user?.username}', does the account own Minecraft Java? Server returned: ${res.statusText}`)
+async function authenticate (client, options) {
+  if (!options.profilesFolder && options.profilesFolder !== false) {
```

Comment:
why not just check `if (!options.profilesFolder)` ?

Reply:
https://github.com/PrismarineJS/node-minecraft-protocol/blob/master/src/client/auth.js#L10

Reply:
I don't get it. The code underneath that sets the folder path to .minecraft is never hit if the option is unspecified (undefined)

Reply:
OK, nevermind, it's set to false. Why this specific exception?

Reply:
Was going for consistency here, but I'll do that over what we have.

Reply:
ok, if there's a valid reason we can keep it. I just wanted some clarification for it, do you know the intention behind it ? The comment doesn't make it entirely clear there.

Reply:
Basically it prevents people from setting profileFolder to true and having it crash.